examples/light-switch-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,150 @@
+OTAImageProcessorImpl gImageProcessor;
+#endif
+
+static const char * TAG = "light-switch-app";
+
+static DeviceCallbacks EchoCallbacks;
+namespace {
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
+app::Clusters::NetworkCommissioning::Instance
+    sWiFiNetworkCommissioningInstance(0 /* Endpoint Id */, &(NetworkCommissioning::ESPWiFiDriver::GetInstance()));
+#endif
+} // namespace
+
+static void InitOTARequestor(void)
+{
+#if CONFIG_ENABLE_OTA_REQUESTOR
+    SetRequestorInstance(&gRequestorCore);
+    gRequestorStorage.Init(Server::GetInstance().GetPersistentStorage());
+    gRequestorCore.Init(Server::GetInstance(), gRequestorStorage, gRequestorUser, gDownloader);
+    gImageProcessor.SetOTADownloader(&gDownloader);
+    gDownloader.SetImageProcessorDelegate(&gImageProcessor);
+    gRequestorUser.Init(&gRequestorCore, &gImageProcessor);
+#endif
+}
+
+static void InitServer(intptr_t context)
+{
+    // Print QR Code URL
+    PrintOnboardingCodes(chip::RendezvousInformationFlags(CONFIG_RENDEZVOUS_MODE));
+
+    static chip::CommonCaseDeviceServerInitParams initParams;
+    (void) initParams.InitializeStaticResourcesBeforeServerInit();
+    chip::Server::GetInstance().Init(initParams);
+
+    // Initialize device attestation config
+    SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
+    sWiFiNetworkCommissioningInstance.Init();
+#endif
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (chip::DeviceLayer::ConnectivityMgr().IsThreadProvisioned() &&
+        (chip::Server::GetInstance().GetFabricTable().FabricCount() != 0))
+    {
+        ESP_LOGI(TAG, "Thread has been provisioned, publish the dns service now");
+        chip::app::DnssdServer::Instance().StartServer();
+    }
+#endif
+    InitBindingHandler();
+}
+
+extern "C" void app_main()
+{
+    // Initialize the ESP NVS layer.
+    esp_err_t err = nvs_flash_init();
+    if (err != ESP_OK)
+    {
+        ESP_LOGE(TAG, "nvs_flash_init() failed: %s", esp_err_to_name(err));
+        return;
+    }
+
+    ESP_LOGI(TAG, "==================================================");
+    ESP_LOGI(TAG, "chip-esp32-light-switch-example starting");
+    ESP_LOGI(TAG, "==================================================");
+
+#if CONFIG_ENABLE_CHIP_SHELL
+    chip::LaunchShell();
+#endif // CONFIG_ENABLE_CHIP_SHELL
+
+    CHIPDeviceManager & deviceMgr = CHIPDeviceManager::GetInstance();
+
+    CHIP_ERROR error = deviceMgr.Init(&EchoCallbacks);
+    if (error != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "device.Init() failed: %s", ErrorStr(error));
+        return;
+    }
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (ThreadStackMgr().InitThreadStack() != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "Failed to initialize Thread stack");
+        return;
+    }
+    if (ThreadStackMgr().StartThreadTask() != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "Failed to launch Thread task");
+        return;
+    }
+#endif
+
+    InitOTARequestor();
```

Comment:
We cannot Initialise OTA requestor before Initialising the Server. Please take a look at changes in #16762 to fix the OTA Requestor initialisation.

Reply:
resolved and rebased